### PR TITLE
Add functionallity for deleting branches.

### LINKED ISSUE #347: Coderadar should delete branch nodes if the branch has been deleted in the remote repository.

### PR DESCRIPTION
**Pull request by maximAtanasov**
If a branch has been deleted in the remote repository, it will now also get deleted from the database.

Closes #347

### CODE REVIEW

File: coderadar-graph/src/main/java/io/reflectoring/coderadar/graph/projectadministration/branch/adapter/DeleteBranchAdapter.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  public void delete(long projectId, Branch branch) {
+    CommitEntity branchCommit = branchRepository.getBranchCommit(projectId, branch.getName());
+    BranchEntity branchEntity =
+        branchRepository.findBranchInProjectByName(projectId, branch.getName());
+    branchRepository.delete(branchEntity);
+    List<BranchEntity> branchesInProject = branchRepository.getBranchesInProject(projectId);
+    deleteCommits(branchCommit, branchesInProject);
+  }
+
+  private void deleteCommits(CommitEntity commit, List<BranchEntity> branchesInProject) {
```
Comment (KilianKrause) [inline coderadar-graph/src/main/java/io/reflectoring/coderadar/graph/projectadministration/branch/adapter/DeleteBranchAdapter.java:39]: Add javadoc.

File: coderadar-test/src/test/java/io/reflectoring/coderadar/rest/branches/DeleteBranchTest.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    projectId =
+        createProjectService.createProject(
+            new CreateProjectCommand(
+                "testProject", null, null, testRepoURL.toString(), false, null, null));
+  }
+
+  @Test
+  void testDeleteBranch() {
+    deleteBranchPort.delete(
+        projectId, new Branch("testBranch2", "fcd9a0e7c34086fdb0aedc82497f8ddfa142e961"));
```
Comment (KilianKrause) [inline coderadar-test/src/test/java/io/reflectoring/coderadar/rest/branches/DeleteBranchTest.java:40]: Create a member for the branch objects.

Comment (maximAtanasov) [reply coderadar-test/src/test/java/io/reflectoring/coderadar/rest/branches/DeleteBranchTest.java:40]: Done.

File: coderadar-graph/src/main/java/io/reflectoring/coderadar/graph/projectadministration/branch/adapter/DeleteBranchAdapter.java (hunk 1 above)
Comment (maximAtanasov) [reply coderadar-graph/src/main/java/io/reflectoring/coderadar/graph/projectadministration/branch/adapter/DeleteBranchAdapter.java:39]: Done.

**Pull request merged**